kombu/transport/redis_cluster.py

```diff
@@ -307,41 +316,79 @@ def _brpop_start(self, timeout=1):
-            conn.client = None
             raise Empty()
-        except MovedError as e:
-            # Copied from redis-py cluster.py
-            self.client.reinitialize_counter += 1
-            if self.client._should_reinitialized():
-                self.client.nodes_manager.initialize()
-                # Reset the counter
-                self.client.reinitialize_counter = 0
-            else:
-                self.client.nodes_manager.update_moved_exception(e)
+        except MovedError:
             raise Empty()
+        conn.client.connection.send_command('BRPOP', conn.key, conn.timeout) # schedule next BRPOP
 
         if resp:
-            dest, item = resp
-            dest = bytes_to_str(dest).rsplit(self.sep, 1)[0]
-            self._queue_cycle.rotate(dest)
-            self.connection._deliver(loads(bytes_to_str(item)), dest)
+            self.deliver_response(resp)
             return True
 
     def _poll_error(self, cmd, conn, **options):
-        if cmd == 'BRPOP':
-            conn.client.parse_response(conn.client.connection, cmd, **options)
+        resp = self.parse_response(conn, 'BRPOP', **options)
+        if resp:
+            self.deliver_response(resp)
+
+        self.connection.cycle._unregister(self, self.client, conn, 'BRPOP')
+
+    def deliver_response(self, resp):
+        dest, item = resp
+        dest = bytes_to_str(dest).rsplit(self.sep, 1)[0]
+        self._queue_cycle.rotate(dest)
+        self.connection._deliver(loads(bytes_to_str(item)), dest)
+
+    def parse_response(self, conn, cmd, **options):
+        try:
+            return conn.client.parse_response(conn.client.connection, cmd, **options)
+        except Exception as e:
+            logger.warning('Error while reading from Redis', extra={"e": e, "key": conn.key})
+            # Mostly copied from https://github.com/sendbird/redis-py/blob/master/redis/cluster.py#L1173
+            if isinstance(e, ConnectionError) or isinstance(e, TimeoutError):
+                self.client.nodes_manager.startup_nodes.pop(target_node.name, None)
+                self.client.nodes_manager.initialize()
+            elif isinstance(e, MovedError):
+                self.client.reinitialize_counter += 1
+                if self.client._should_reinitialized():
+                    self.client.nodes_manager.initialize()
+                    self.client.reinitialize_counter = 0
+                else:
+                    self.client.nodes_manager.update_moved_exception(e)
+            elif isinstance(e, SlotNotCoveredError):
+                self.client.reinitialize_counter += 1
+                if self.client._should_reinitialized():
+                    self.client.nodes_manager.initialize()
+                    self.client.reinitialize_counter = 0
+            elif isinstance(e, TryAgainError):
+                return  # try again in next BRPOP
+            elif isinstance(e, AskError):
+                self.add_ask_error(e, conn)
```

Comment:
I don't get the point why you stored this error information. you do not use exception information that indicates you need to connect other node.